src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs

```diff
@@ -275,24 +277,65 @@ protected override sealed void ProcessRecord()
         #region Cmdlets base APIs
 
         /// <summary>
-        /// Initializes source repositories for PowerShell cmdlets, based on config, source string, and/or host active source property value.
+        /// Looks through all available sources (including those disabled) by matching source name and url to get matching sources.
         /// </summary>
         /// <param name="source">The source string specified by -Source switch.</param>
-        protected void UpdateActiveSourceRepository(string source)
+        /// <returns></returns>
+        protected PackageSource GetMatchingSource(string source)
         {
             var packageSources = _sourceRepositoryProvider?.PackageSourceProvider?.LoadPackageSources();
+            var matchingSource = packageSources
+                ?.Where(p => StringComparer.OrdinalIgnoreCase.Equals(p.Name, source) ||
+                             StringComparer.OrdinalIgnoreCase.Equals(p.Source, source))
+                .FirstOrDefault();
```

Comment:
@mishra14 The motivation was to verify whether existing code has bugs or not.
Anyway I just ran a simple test and verified it works fine when `packageSource` is null.

Although this linq expression is not optimal, it can be simplified to
```c#
var matchingSource = packageSources
    ?.FirstOrDefault(p => StringComparer.OrdinalIgnoreCase.Equals(p.Name, source) ||
                          StringComparer.OrdinalIgnoreCase.Equals(p.Source, source));
```
I'll let you decide whether or not to take it :smiley_cat: